### DISCUSSION TITLE
How to increase TPM, Hammerdb ?

### DISCUSSION THREAD

**Discussion by springeee** [TPC-Council/HammerDB#157] [Q&A] [answered]
Hi all:

    use Hammerdb test Oracle Database .

after schema build , run the test .I found that the first time run 1 virtual user the TPM is 104335 and use "iostat -x 1" see the disk io , all the io is write. the second , third time run 1 virtual use the TPM is  77468 and 68087 . and the disk io , have write and read IO.
1, shall I rebuild schema when run the test every time ?
2,  when run the test ,  CPU , memory and disk are idle,  how to increase TPM ?
3,  the Hammerdb TPC-C test, what's the IO model ? full write or mix write and read ?

Thanks,
Regards
Springee

**Comment by sm-shaw** [comment]
No you do not need to rebuild the schema after each test. 
I recommend using the built-in Oracle AWR report generator and look at the top timed events first. Focus on where Oracle is spending its time if you want to improve performance.

**Comment by springeee** [comment]
> No you do not need to rebuild the schema after each test. - -ok
> I recommend using the built-in Oracle AWR report generator and look at the top timed events first. Focus on where Oracle is spending its time if you want to improve performance. -- yes, i have use AWR

![image](https://user-images.githubusercontent.com/72604063/95965426-80ffb380-0e3c-11eb-91ef-d1b7394d573d.png)

**Comment by sm-shaw** [comment, marked as answer]
It appears that you are now able to run HammerDB successfully against an Oracle database. Therefore we are now leaving the subject area where HammerDB support is applicable and instead you should focus on Oracle database performance tuning and optimization support forums and documentation for transactional workloads. There are also many good books on Oracle database performance such as the ones published by Apress https://www.apress.com/us/product-search?query=oracle.  As a starting point the database is spending almost 24% of its time on "log file sync" and therefore redo log configuration would be a great place for you to start.

**Comment by springeee** [comment]
Hi Steve:
    Thanks for your advice.
Regards.
